pkg/util/validation/limits.go

```diff
@@ -885,3 +895,100 @@ func EnabledByAnyTenant(tenantIDs []string, f func(string) bool) bool {
 	}
 	return false
 }
+
+// MustRegisterExtension registers the extensions type with given name
+// and returns a function to get a pointer to the extensions E from a *Limits instance.
+// The name will be used as YAML/JSON key to decode the extensions.
+// This method is not thread safe and should be called only during package initialization.
+func MustRegisterExtension[E any](name string) func(*Limits) *E {
+	for _, e := range registeredExtensions {
+		if e.name == name {
+			panic(fmt.Errorf("extension %s already registered", name))
+		}
+	}
+
+	extensionsType := reflect.TypeOf(new(E))
+	registeredExtensions = append(registeredExtensions, extension{name: name, typ: extensionsType})
+
+	return func(l *Limits) *E {
+		if e, ok := l.extensions[name]; ok {
+			return e.(*E)
+		}
+		return nil
+	}
+}
+
+type extension struct {
+	name string
+	typ  reflect.Type
+}
+
+// registeredExtensions is the list of registered extensions
+var registeredExtensions []extension
+
+// newExtensionsOnlyLimitsConfig returns an interface{} value of a pointer to a struct of type:
+//
+//	struct {
+//	    ExtName1 T1 `yaml:"extname1"`
+//	    ...
+//	    ExtN TN `yaml:"extN"`
+//	    Throwaway map[string]interface{} `yaml:",inline"`
```

Comment:
My understanding is that we need this so that we can first unmarshal the extension by itself. The `Throwaway` will take all the fields of the plain limits config. Is this right? And can you document it?

Reply:
Also, shouldn't this also have `json:",inline"`?

Reply:
Documented it. Also documented that `json:",inline"` does not exist, we just embed the field to achieve the same thing.